taskweaver/llm/google_genai.py

```diff
@@ -147,12 +147,14 @@ def _chat_completion(
                 raise Exception(f"Invalid role: {msg['role']}")
 
         if stream is False:
-            response = self.model.generate_content(genai_messages, stream=False)
+            response:GenerateContentResponse = self.model.generate_content(genai_messages, stream=False)
             yield format_chat_message("assistant", response.text)
-
-        response = self.model.generate_content(genai_messages, stream=True)
```

Comment:
The original code handles the case when stream is true from here.

Reply:
Yes, stream=True is handled fine but stream=False will cause both false and true to be executed because there is no else.
`
```python
 if stream is False:
            response = self.model.generate_content(genai_messages, stream=False)
            yield format_chat_message("assistant", response.text)

        response = self.model.generate_content(genai_messages, stream=True)
        for chunk_obj in response:
            yield format_chat_message("assistant", chunk_obj.text)
```

Reply:
Have you tested when stream==False? when stream==False, the response will contain all the generated content and the first `yield` would return all the content at once and will not reach the following part, right?

Reply:
A better way is to turn the first `yield` to `return`?